src/Std/Data/DTreeMap/Internal/Zipper.lean

```diff
@@ -0,0 +1,1152 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Wojciech Różowski
+-/
+module
+
+prelude
+public import Std.Data.Iterators.Lemmas.Producers.Slice
+public import Init.Data.Slice
+public import Std.Data.DTreeMap.Internal.Lemmas
+
+namespace Std.DTreeMap.Internal
+open Std.Iterators
+
+section Zipper
+public inductive Zipper (α : Type u) (β : α → Type v) where
+  | done
+  | cons (k : α) (v : β k) (tree : Impl α β) (next : Zipper α β)
+
+variable {α : Type u} {β : α → Type v}
+public def Zipper.toList : Zipper α β → List ((a : α) × β a)
+| .done => []
+| .cons k v tree next => ⟨k,v⟩ :: tree.toList ++ next.toList
+
+public def Zipper.Ordered [Ord α] (t : Zipper α β) : Prop :=
+  t.toList.Pairwise (fun a b => compare a.1 b.1 = .lt)
+
+def Zipper.size : Zipper α β → Nat
+| .done => 0
+| .cons _ _ tree next => 1 + tree.treeSize + next.size
+
+public def Zipper.prependMap : Impl α β → Zipper α β → Zipper α β
+  | .leaf, it => it
+  | .inner _ k v l r, it => prependMap l (.cons k v r it)
+
+public def Zipper.prependMapGE [Ord α] (t : Impl α β) (lower_bound : α)
```

Comment:
Please use `lowerCamelCase` for arguments which are data, like `lower_bound`. This makes it easier to tell at a glance that this is data and not a proof.